### PR TITLE
Adding descriptions in HttpSysOptions

### LINKED ISSUE #28297: Improve HttpSysOptions descriptions
The concept docs for HttpSys have some better descriptions of the HttpSys options than the API docs, especially the default values. 

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/httpsys?view=aspnetcore-5.0#configure-the-aspnet-core-app-to-use-httpsys
https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.httpsys.httpsysoptions?view=aspnetcore-5.0
https://github.com/dotnet/aspnetcore/blob/58a7eb63ad5c6b037c43d2bd178c79f4248bb71f/src/Servers/HttpSys/src/HttpSysOptions.cs

We don't want to maintain two copies of these options docs, we should improve the API docs and remove the redundant entries from the concept docs.

See https://github.com/dotnet/AspNetCore.Docs/issues/20827

### PR DESCRIPTION
**Pull request by utsabshrestha**
# Improve HttpSysOptions descriptions

- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

## Description

Added the descriptions in the HttpSysOptions from https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/httpsys?view=aspnetcore-5.0#configure-the-aspnet-core-app-to-use-httpsys to improve the API docs.

Fixes #28297

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=37949) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=37949) before we can accept your contribution.<br/><br/>:x: utsabshrestha [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=37949)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=37949) it.</sub>

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 1]
```diff
@@ -69,24 +71,26 @@ public HttpSysOptions()
         public ClientCertificateMethod ClientCertificateMethod { get; set; } = ClientCertificateMethod.AllowCertificate;
 
         /// <summary>
-        /// Gets or sets the number of concurrent workers draining requests from the Http.sys queue.
+        /// The maximum number of concurrent accepts.
+        /// The default is 5 times the number of processors as returned by <see cref="Environment.ProcessorCount" />.
         /// </summary>
         /// <remarks>
         /// Defaults to 5 times the number of processors as returned by <see cref="Environment.ProcessorCount" />.
         /// </remarks>
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs:79]:
```suggestion
```

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 2]
```diff
@@ -69,24 +71,26 @@ public HttpSysOptions()
...
         /// </summary>
         public bool EnableResponseCaching { get; set; } = true;
 
         /// <summary>
-        /// The url prefixes to register with Http.Sys. These may be modified at any time prior to disposing
-        /// the listener.
-        /// When attached to an existing queue the prefixes are only used to compute PathBase for requests.
+        /// Specify the UrlPrefixCollection to register with HTTP.sys.
+        /// The most useful is UrlPrefixCollection.Add, which is used to add a prefix to the collection.
+        /// These may be modified at any time prior to disposing the listener.
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs:93]: Revert, the prior comments made more sense.

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 3]
```diff
@@ -104,15 +108,15 @@ public HttpSysOptions()
         public TimeoutManager Timeouts { get; } = new TimeoutManager();
 
         /// <summary>
-        /// Gets or Sets if response body writes that fail due to client disconnects should throw exceptions or
-        /// complete normally. The default is false.
+        /// Indicate if response body writes that fail due to client disconnects should throw exceptions or complete normally.
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs:111]: Revert, `Gets or Sets` is standard terminology for properties. Please use it for the remaining properties as well.

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 4]
```diff
@@ -39,6 +39,7 @@ public HttpSysOptions()
 
         /// <summary>
         /// The name of the Http.Sys request queue
+        /// The default is `null (Anonymous queue)`.
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs:42]: The (description) should be outside of the code block in all of these comments.

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 5]
```diff
@@ -104,15 +108,15 @@ public HttpSysOptions()
...
+        /// The default is `false (complete normally)`.
         /// </summary>
         public bool ThrowWriteExceptions { get; set; }
 
         /// <summary>
-        /// Gets or sets the maximum number of concurrent connections to accept, -1 for infinite, or null to
-        /// use the machine wide setting from the registry. The default value is null.
-        /// This settings does not apply when attaching to an existing queue.
+        /// The maximum number of concurrent connections to accept. Use -1 for infinite.
+        /// Use null to use the registry's machine-wide setting.
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs:118]:
```suggestion
        /// Set to `null` to use the registry's machine-wide setting.
```

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 6]
```diff
@@ -104,15 +108,15 @@ public HttpSysOptions()
...
+        /// Indicate if response body writes that fail due to client disconnects should throw exceptions or complete normally.
+        /// The default is `false (complete normally)`.
         /// </summary>
         public bool ThrowWriteExceptions { get; set; }
 
         /// <summary>
-        /// Gets or sets the maximum number of concurrent connections to accept, -1 for infinite, or null to
-        /// use the machine wide setting from the registry. The default value is null.
-        /// This settings does not apply when attaching to an existing queue.
+        /// The maximum number of concurrent connections to accept. Use -1 for infinite.
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs:117]:
```suggestion
        /// Gets or sets the maximum number of concurrent connections to accept. Set `-1` for infinite.
```

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 7]
```diff
@@ -134,8 +138,8 @@ public HttpSysOptions()
         }
 
         /// <summary>
-        /// Gets or sets the maximum number of requests that will be queued up in Http.Sys.
-        /// This settings does not apply when attaching to an existing queue.
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs:138]: Keep this comment about existing queues.

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 8]
```diff
@@ -164,6 +168,7 @@ public long RequestQueueLimit
         /// When set to null, the maximum request body size is unlimited.
         /// This limit has no effect on upgraded connections which are always unlimited.
         /// This can be overridden per-request via <see cref="IHttpMaxRequestBodySizeFeature"/>.
+        /// Defaults to 30,000,000 bytes, which is approximately 28.6MB.
         /// </summary>
         /// <remarks>
         /// Defaults to 30,000,000 bytes, which is approximately 28.6MB.
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs:174]: Duplicate, remove the remarks section.

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 9]
```diff
@@ -182,16 +187,14 @@ public long RequestQueueLimit
...
         /// </summary>
         public bool AllowSynchronousIO { get; set; }
 
         /// <summary>
-        /// Gets or sets a value that controls how http.sys reacts when rejecting requests due to throttling conditions - like when the request
-        /// queue limit is reached. The default in http.sys is "Basic" which means http.sys is just resetting the TCP connection. IIS uses Limited
-        /// as its default behavior which will result in sending back a 503 - Service Unavailable back to the client.
-        /// This settings does not apply when attaching to an existing queue.
+        /// The HTTP.sys behavior when rejecting requests due to throttling conditions.
+        /// The devault is <see cref="Http503VerbosityLevel"/>.Basic .
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs:197]: Revert, the original comment was more descriptive.

**Comment by rafikiassumani-msft** [comment]
@utsabshrestha we will close this PR for now as you have not made the requested changes yet. We will be happy to re-open this once you make the changes. Thank you.

**Pull request closed without merging**